### PR TITLE
Fix wrong seasonality basis generator for NBEATS

### PR DESCRIPTION
**Pull request by vladimir-chernykh**
Fixes https://github.com/unit8co/darts/issues/803.

### Summary

It fixes wrong seasonality basis vectors generated for NBEATS interpretable model. Now the basis vectors are normalized by the horizon length as it is done in the paper.

### Other Information

--

### CODE REVIEW

**Review by dennisbader — COMMENTED** [review]
Hi @vladimir-chernykh and thanks a lot for pointing this out!

I checked and agree with you. That one was easy to overlook!

I initiated the tests

File: darts/models/forecasting/nbeats.py [hunk 1]
```diff
@@ -50,11 +50,11 @@ def __init__(self, target_length):
...
         half_minus_one = int(target_length / 2 - 1)
         cos_vectors = [
-            torch.cos(torch.arange(target_length) * 2 * np.pi * i)
+            torch.cos(torch.arange(target_length) / target_length * 2 * np.pi * i)
             for i in range(1, half_minus_one + 1)
         ]
         sin_vectors = [
-            torch.sin(torch.arange(target_length) * 2 * np.pi * i)
+            torch.sin(torch.arange(target_length) / target_length * 2 * np.pi * i)
             for i in range(1, half_minus_one + 1)
```
Comment (hrzn) [inline darts/models/forecasting/nbeats.py:58]: Good catch!
Now, looking again at Eq (3) in the paper, I'm wondering whether we shouldn't also change the range of `i`, as it should start from 0 for the cosines and go from ceil(H/2) to ceil(H-1) for the sines.

So something like changing the range to this for both sine and cosine:
```
for i in range(0, half_minus_one)
```
for i in range(half_minus_one + 1, 
```

And adding `target_length / 2` to `i` in the sines loop.

Unless I'm missing something?

**Comment by vladimir-chernykh** [comment]
@hrzn,

I believe the range of `i` should start from 0 for both sines and cosines.

Screenshot 2022-02-17 at 12 32 22

What you're talking about I think is the theta coefficients. For cosines we take first floor(H/2-1) values from theta vector and for sines we take next floor(H/2-1) values starting from floor(H/2). But this is already done correctly in the DARTS. So in my opinion no more changes needed here.

**Comment by hrzn** [comment]
> @hrzn,
> 
> I believe the range of `i` should start from 0 for both sines and cosines.
> 
> Screenshot 2022-02-17 at 12 32 22
> 
> What you're talking about I think is the theta coefficients. For cosines we take first floor(H/2-1) values from theta vector and for sines we take next floor(H/2-1) values starting from floor(H/2). But this is already done correctly in the DARTS. So in my opinion no more changes needed here.

You are right - I convinced myself too between my earlier comment and now. Thanks for checking - and good catch! We will release it (along with a planned patch release) asap.

**Pull request merged**